src/test/java/org/sagebionetworks/bridge/spring/controllers/HealthDataDocumentationControllerTest.java

```diff
@@ -0,0 +1,130 @@
+import org.sagebionetworks.bridge.models.HealthDataDocumentation;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.sagebionetworks.bridge.models.StatusMessage;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.services.HealthDataDocumentationService;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.sagebionetworks.bridge.BridgeConstants.API_DEFAULT_PAGE_SIZE;
+import static org.sagebionetworks.bridge.TestConstants.IDENTIFIER;
+import static org.sagebionetworks.bridge.TestConstants.TEST_APP_ID;
+import static org.sagebionetworks.bridge.TestUtils.assertCrossOrigin;
+import static org.sagebionetworks.bridge.TestUtils.assertDelete;
+import static org.sagebionetworks.bridge.TestUtils.assertGet;
+import static org.sagebionetworks.bridge.TestUtils.assertPost;
+import static org.sagebionetworks.bridge.TestUtils.mockRequestBody;
+import static org.sagebionetworks.bridge.models.ResourceList.OFFSET_KEY;
+import static org.testng.Assert.assertSame;
+import static org.testng.AssertJUnit.assertEquals;
+
+public class HealthDataDocumentationControllerTest {
+    private static final String TEST_DOCUMENTATION = "test documentation";
+
+    @Mock
+    private HealthDataDocumentationService mockService;
+
+    @InjectMocks
+    @Spy
+    private HealthDataDocumentationController controller;
+
+    @Mock
+    private HttpServletRequest mockRequest;
+
+    @BeforeMethod
+    public void before() {
+        MockitoAnnotations.initMocks(this);
+
+        // Mock request
+        doReturn(mockRequest).when(controller).request();
+
+        // Mock session
+        UserSession mockSession = new UserSession();
+        mockSession.setAppId(TEST_APP_ID);
+        doReturn(mockSession).when(controller).getAuthenticatedSession(Roles.RESEARCHER, Roles.DEVELOPER);
+    }
+
+    @Test
+    public void verifyAnnotations() throws Exception {
+        assertCrossOrigin(HealthDataController.class);
+        assertPost(HealthDataDocumentationController.class, "createOrUpdateHealthDataDocumentation");
+        assertGet(HealthDataDocumentationController.class, "getHealthDataDocumentationForId");
+        assertGet(HealthDataDocumentationController.class, "getAllHealthDataDocumentationForParentId");
+        assertDelete(HealthDataDocumentationController.class, "deleteHealthDataDocumentationForIdentifier");
+        assertDelete(HealthDataDocumentationController.class, "deleteAllHealthDataDocumentationForParentId");
+    }
+
+    @Test
+    public void createOrUpdateHealthDataDocumentation() throws Exception {
+        HealthDataDocumentation doc = HealthDataDocumentation.create();
+        doc.setParentId(TEST_APP_ID);
```

Comment:
Instead of setting the parentId as part of the input, we should leave this blank and verify that we get the correct parentId from the session.

Reply:
@DwayneJengSage I have been stuck on this one - I can't seem to get an argument captor working to see that the healthDataDocumentation was set properly. It seems simple but I am still struggling with Mockito. Could you please help point me in a better direction?

Reply:
Is this one of those situations where I should create a helper function in the controller in order to simplify verifying the arguments, similar to queryHelper in various DAO's?

Reply:
I figured it out with the above method!